gems/pending/util/MiqSockUtil.rb

```diff
@@ -8,6 +8,10 @@ def self.getHostName
   # Return the cannonical name for host in DNS
   def self.getFullyQualifiedDomainName
     Socket.gethostbyname(Socket.gethostname).first
+  rescue => e
+    $log.error "The appliance hostname could not be resolved. Please make sure to set a valid fully qualified hostname "\
+               "for SmartState analysis to work correctly."
+    raise e
```

Comment:
You can just call `raise` with no parameters and it will re-raise the current Exception.  This is a preferable way to reraise.